server/rootvalue/storage.go

```diff
@@ -0,0 +1,313 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rootvalue
+
+import (
+	"context"
+	"fmt"
+
+	doltserial "github.com/dolthub/dolt/go/gen/fb/serial"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/store/hash"
+	"github.com/dolthub/dolt/go/store/prolly"
+	"github.com/dolthub/dolt/go/store/prolly/shim"
+	"github.com/dolthub/dolt/go/store/prolly/tree"
+	"github.com/dolthub/dolt/go/store/types"
+	flatbuffers "github.com/dolthub/flatbuffers/v23/go"
+
+	"github.com/dolthub/doltgresql/flatbuffers/gen/serial"
+)
+
+type fbRvStorage struct {
```

Comment:
Might figure out how to better share code with Dolt here. The table maps are identical between the two root values down their serialization, would be nice to have a single library that reads and writes them in both implementations.